docs/components/Video/Video.js

```diff
@@ -10,26 +11,68 @@ export const Video = ({
   height = 315,
 }) => {
   const videoId = getUrlParamsFromString(url).find(({ key }) => key === 'v')?.value;
+  const [isVideoVisible, setIsVideoVisible] = useState(false);
+  const videoContainerRef = useRef();
+
+  useEffect(() => {
+    const options = {
+      root: null,
+      rootMargin: '0px',
+      threshold: 0.5, // Adjust this threshold as needed
+    };
+
+    let observer;
+    
+    if (videoContainerRef.current) {
+      observer = new IntersectionObserver(handleIntersection, options);
+      observer.observe(videoContainerRef.current);
+    }
+
+    return () => {
+      if (observer && videoContainerRef.current) {
+        observer.unobserve(videoContainerRef.current);
+      }
+    };
+  }, []);
+
+  const handleIntersection = (entries) => {
+    entries.forEach((entry) => {
+      if (entry.isIntersecting) {
+        setIsVideoVisible(true);
+      }
+    });
+  };
 
   return (
     <figure className={styles.video}>
-      <div
-        className={styles.videoContainer}
-        style={{
-          paddingTop: `${(height / width) * 100}%`,
-        }}
-      >
-        <iframe
-          title={title}
-          width={width}
-          height={height}
-          src={`https://www.youtube.com/embed/${videoId}?feature=oembed`}
-          frameBorder="0"
-          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
-          allowFullScreen
-          autoPlay
-        />
-      </div>
+      {isVideoVisible ? (
+        <div
+          ref={videoContainerRef}
+          className={styles.videoContainer}
+          style={{
+            paddingTop: `${(height / width) * 100}%`,
+          }}
+        >
+          <iframe
+            title={title}
+            width={width}
+            height={height}
+            src={`https://www.youtube.com/embed/${videoId}?feature=oembed`}
+            frameBorder="0"
+            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+            allowFullScreen
+            autoPlay
+          />
+        </div>
+      ) : (
+        <div
+          ref={videoContainerRef}
+          style={{
+            width: '100%',
```

Comment:
just to be clear - were you saying on the other Pr that this should _not_ be the following?

```
style={{
            paddingTop: `${(height / width) * 100}%`,
          }}
```
i would expect the container to not take the right amount of space without that when the video hasn't loaded

Reply:
i suspect that the visibility updates so fast, with the CSS rendering the visible space, that we dont even notice it, but i might be wrong, because before it's loaded, it definitely takes up 0 height:

<img width="1499" alt="image" src="https://github.com/colbyfayock/next-cloudinary/assets/1045274/308536b2-3bad-40a8-9bc6-54d08af4875f">

i tried even throttling and its very fast to load in

it _is_ changing the page height, but its just not visible. idk if that has any implications though or not

would you mind just updating it to the `paddingTop`? feel a bit safer that way and feel like it wouldnt do any harm

then we can merge it in

Reply:
did it @colbyfayock